File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        while (true) {
            String input = br.readLine();
            if (input.equals("0")) break;

            int length = input.length();
            String answer = "yes";
            for (int i = 0; i < length / 2; i++) {
                if (input.charAt(i) != input.charAt(length - i - 1)) {
                    answer = "no";
                    break;
                }
            }
            sb.append(answer).append("\n");
        }
        System.out.print(sb);
    }
}

File: Python/백준/Bronze/1259. 팰린드롬수/팰린드롬수.py
while True:
    data = input()
    if int(data) == 0: break

    answer = "yes"
    for i in range(len(data)//2):
        left = i
        right = len(data) - i - 1
        if data[left] != data[right]:
            answer = "no"
            break

    print(answer)